src/Rules.UnitTests/LocalRuleMetadataProviderTests.cs

```diff
@@ -88,12 +88,22 @@ void CheckRule(string fullResourceName)
                 Console.WriteLine("Checking " + fullResourceName);
 
                 SonarCompositeRuleId ruleId = GetCompositeRuleIdFromResourceName(fullResourceName);
-                var expectedDescription = GetEmbeddedRuleDescription(fullResourceName);
+                var expectedRuleInfo = GetEmbeddedRuleDescription(fullResourceName);
 
                 var actual = testSubject.GetRuleInfo(ruleId);
 
                 actual.FullRuleKey.Should().Be(ruleId.ToString());
-                actual.Description.Should().Be(expectedDescription);
+                actual.Description.Should().Be(expectedRuleInfo.Description);
+                
+                if (actual.DescriptionSections.Count > 0)
+                {
+                    actual.DescriptionSections.Should().BeEquivalentTo(expectedRuleInfo.DescriptionSections);
+                    actual.DescriptionSections.All(section => !string.IsNullOrWhiteSpace(section.HtmlContent)).Should().BeTrue();
+                }
+
+                (!string.IsNullOrWhiteSpace(actual.Description) || actual.DescriptionSections.Count > 0)
```

Comment:
Just to be clear, is it ok for a rule to have a non-null `Description` and multiple `DescriptionSections`?

Reply:
Yes. Some rule descriptions still have legacy rule description along with the rich